src/app/berkeley_migration/berkeley_migration.ml

```diff
@@ -142,7 +143,7 @@ let user_commands_from_block_id ~mainnet_pool block_id =
           ~f:mainnet_transaction_failure_of_string
       in
       let hash =
-        Mina_transaction.Transaction_hash.of_base58_check_exn user_cmd.hash
+        Mina_transaction.Transaction_hash.of_base58_check_exn_v1 user_cmd.hash
```

Comment:
I think you should also change in archive_lib.user_commands.add_extensional_if_doesn't_exist function. Otherwise the hash you writen to the migrated database would be changed: https://github.com/MinaProtocol/mina/blob/79c6253989ddaa0db2c4c7267f3dc1a5bf9f12e9/src/app/archive/lib/processor.ml#L1882

I am not sure

Reply:
I sent an email to Deepthi with a suggested way to handle this issue. You'd add an optional boolean flag to `add_extensional_if_doesn't_exist` that would indicate whether to use the `V1` hash.